app.rb

```diff
@@ -0,0 +1,179 @@
+      print "\n"
+      age = age.to_i
+      break if name.length.positive? && specialization.length.positive? && age.positive?
+
+      puts 'Wrong input'
+    end
+    puts "Teacher entered.. #{name}  #{age} #{specialization}"
+    Teacher.new(specialization, age, name, true)
+  end
+
+  def create_student
+    puts 'Creating student...'
+    name = '', age = 0, parent_permission = false
+    loop do
+      print 'Name: '
+      name = gets.chomp
+      print "\n"
+      print 'Age: '
+      age = gets.chomp
+      age = age.to_i
+      print "\n"
+      print 'Has parent permission [Y/N]: '
+      answer = gets.chomp
+      parent_permission = true if answer.upcase == 'Y' && age > 18
+      print "\n"
+      break if name.length.positive? && [true, false].include?(parent_permission) && age.positive?
+
+      puts 'Wrong input: Try again *'
+    end
+    puts "Student entered.. #{name}  #{age} #{parent_permission}"
+    Student.new(name, age, parent_permission)
+  end
+
+  def create_person
+    loop do
+      print "\n"
+      print 'Do you want to create a student(1) or teacher(2)? [input number]: '
+      answer = gets.chomp
+      answer = answer.to_i
+      person = nil
+      case answer
+      when 1
+        person = create_student
+      when 2
+        person = create_teacher
+      else
+        puts 'Wrong input'
+      end
+      next unless person
+
+      persons.push(person)
+      puts 'Person created successfully'
+      break
+    end
+  end
+
+  def create_book
+    puts 'Creating book...'
+    title = '', author = ''
+    loop do
+      print 'Title: '
+      title = gets.chomp
+      print "\n"
+      print 'Author: '
+      author = gets.chomp
+      print "\n"
+      break if title.length.positive? && author.length.positive?
+
+      puts 'Wrong input: Try again *'
+    end
+    puts "Book entered.. #{title}  #{author}"
+    book = Book.new(title, author)
+    books.push(book)
+  end
+
+  def create_rental
+    return if persons.empty? || books.empty?
+
+    date = '', person_index = 0, book_index = 0
+    loop do
+      list_people
+      puts 'Select a person from persons listed by number: '
+      person_index = parse_int
+      list_books
+      puts 'Select a book from books listed by number: : '
+      book_index = parse_int
+      puts 'Enter date : [2022-07-24 | yy.mm.dd]'
+      date = parse_date
+      break if person_index.positive? && book_index.positive? && date
+
+      puts 'Wrong input: Try again *'
+    end
+    person = persons[person_index - 1]
+    book = books[book_index - 1]
+    rental = Rental.new(date, person, book)
+    rentals.push(rental)
+  end
+
+  def list_rentals
+    selcted_person = nil, id = 0
+    list_people
+    loop do
+      print 'Enter ID of person from listed persons'
+      id = parse_int
+      persons.each do |person|
+        if person.id == id
+          selcted_person = person
+          break
+        end
+      end
+      next unless selcted_person
```

Comment:
- [ ] I am getting an error when I try to List all rentals for a given id'
![6](https://user-images.githubusercontent.com/44745855/181918676-780e9aa5-6c62-4916-9b42-794f3776608c.png)

Reply:
Thanks @DeliceLydia  required changes made 👍🏽 
![image](https://user-images.githubusercontent.com/94127418/181925346-c714094b-7ba9-47ae-ae8a-7ee30ba05ff1.png)